### PR TITLE
Change canvas change trigger from a mutation observer to an event listener

### PR DESCRIPTION
**Pull request by masaball**
This PR adjusts how `canvasIndex` in the state is set. The behavior with these changes is somewhat flaky. Sometimes the hooks fire on every canvas change, sometimes they don't fire at all. I would guess it has something to do with the event listener not actually getting associated with player. 

Having the event listener and the mutation observer in place at the same time I think did provide somewhat more consistent behavior in testing when switching canvases but would trigger the transcript component to enter its loading state multiple times.

@Dananji, could you do some testing and see if you can get it working consistently?

### CODE REVIEW

**Comment by Dananji** [comment]
The reason the transcript switching broke when there is a blank Canvas is because the `MutationObserver` we are using in the Transcript component to observe the player changes lost its connection to the player instance when we load an empty Canvas into the player without a `src` attribute.
The solution to make this work: load the hidden VideoJS player with a sample `src` attribute. This can be either a URL to an external resource or an absolute path to a local resource within Ramp. Using an external URL doesn't give Ramp the control of the resource, which could be an issue. So resolving to use a local resources requires an investigation and an implementation of Rollup configuration of how to bundle that into Ramp without ballooning the package size. 

The quick and dirty solution in this PR is to use a time interval running continuously to poll the player instance on the page, and read its `data-canvasindex` attribute to update the state in the Transcript component to load relevant transcripts into the UI.

File: src/components/Transcript/Transcript.js [hunk 1]
```diff
@@ -33,14 +33,13 @@ const Transcript = ({ playerID, manifestUrl, transcripts = [] }) => {
     isMachineGen: false,
   });
   const [canvasIndex, _setCanvasIndex] = React.useState(0);
+  const [canvasIsEmpty, setCanvasIsEmpty] = React.useState(false);
```
Comment (masaball) [inline src/components/Transcript/Transcript.js:36]: With the logic to explicitly update this being removed does this actually do anything? Is there background processing that happens that triggers this hook?

**Review by masaball — COMMENTED** [review]
Looks good to me, though I had a question about the `canvasIsEmpty` hook.

File: src/components/Transcript/Transcript.js (hunk 1 above)
Comment (Dananji) [reply src/components/Transcript/Transcript.js:36]: No, it doesn't. Good catch!
This can be removed.

**Comment by cjcolvar** [comment]
As far as I can tell this looks good to me.  With the `setInterval` polling do we not need a `src` for the hidden video.js player?

**Review by cjcolvar — APPROVED** [review]

**Pull request merged**